### PR TITLE
[Docs] Update writing templates to include arrays + json input step

### PR DESCRIPTION
**Pull request by anovis**
## Hey, I just made a Pull Request!

Updating documentation to include information on templating arrays as well as the correct syntax to pass json as inputs to a template step. Based on discussion in discord.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/features/software-templates/writing-templates.md [hunk 1]
```diff
@@ -304,7 +321,8 @@ You might have noticed in the examples that there are `{{ }}`, and these are a
 `yaml` together. All the form inputs from the `parameters` section, when passed
 to the steps will be available by using the template syntax
 `{{ parameters.something }}`. This is great for passing the values from the form
-into different steps and reusing these input variables.
+into different steps and reusing these input variables. To pass arrays or objects
+use the syntax `{{ json paramaters.something }}`.
```
Comment (benjdlambert) [inline docs/features/software-templates/writing-templates.md:325]: Maybe just clarify that `parameters.something` should be of type `object` or `array` in the `jsonSchema` here too?

Comment (benjdlambert) [reply docs/features/software-templates/writing-templates.md:325]: Like `nicknames` above for example. Or something like that?

File: docs/features/software-templates/writing-templates.md [hunk 2]
```diff
@@ -304,7 +321,10 @@ You might have noticed in the examples that there are `{{ }}`, and these are a
 `yaml` together. All the form inputs from the `parameters` section, when passed
 to the steps will be available by using the template syntax
 `{{ parameters.something }}`. This is great for passing the values from the form
-into different steps and reusing these input variables.
+into different steps and reusing these input variables. To pass arrays or objects
+use the syntax `{{ json paramaters.something }}` where `paramaters.something` is 
+of type `object` or `array` in the jsonSchema, such as the nicknames paramter
```
Comment (benjdlambert) [inline docs/features/software-templates/writing-templates.md:326]:
```suggestion
of type `object` or `array` in the `jsonSchema`, such as the `nicknames` parameter 
```

**Comment by benjdlambert** [comment]
Looks like you need to run prettier on the docs @anovis

**Review by freben — APPROVED** [review]

**Pull request merged**